update-asdf-and-dockerfile/action.yml

```diff
@@ -114,6 +114,14 @@ runs:
         asdf install python && \
         asdf install poetry && \
         poetry lock --no-update
+    - name: Update latest poetry version to pyproject.toml if it exists
+      if: ${{ inputs.plugin == 'poetry' && env.CURRENT_VERSION != env.LATEST_VERSION }}
+      shell: bash
+      run: |
+        test -f pyproject.toml && \
+        sed -i "s/^poetry =.*/poetry = \"^${{ env.LATEST_VERSION }}\"/" pyproject.toml && \
```

Comment:
So there isn't a way to do this directly with poetry?

Are you sure this value doesn't get updated already in `poetry lock --no-update`?

Reply:
I'll double check this!

Reply:
I double check that this is not updating poetry, but there is `poetry version` that does: https://python-poetry.org/docs/cli/#version

Reply:
This seems to work perfectly:
```
$ poetry version 1.5.1
Bumping version from 1 to 1.5.1
$ git diff
diff --git a/pyproject.toml b/pyproject.toml
index e1bba59e7..8e4f442b3 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "internal-project"
-version = "1"
+version = "1.5.1"
```

So we should be able to use this instead of sed 🤔